presto-common/src/main/java/com/facebook/presto/common/RuntimeStats.java

```diff
@@ -74,15 +75,18 @@ public RuntimeMetric getMetric(String name)
     }
 
     @JsonValue
-    public Map<String, RuntimeMetric> getMetrics()
+    public ConcurrentHashMap<String, RuntimeMetric> getMetrics()
     {
         return metrics;
     }
 
     public void addMetricValue(String name, long value)
     {
         metrics.computeIfAbsent(name, RuntimeMetric::new);
-        metrics.get(name).addValue(value);
+        metrics.computeIfPresent(name, (metricName, metric) -> {
+            metric.addValue(value);
+            return metric;
+        });
```

Comment:
Some of this code is coming from the fact that `RuntimeMetric` is not thread safe. In `RuntimeMetric` we should make the following changes to make that class multi-threaded ready - 

1. Add `volatile` to all non-final member variables to allow for thread visibility
2. Add `synchronized` to `addValue` and `mergeWith` functions so that the other use cases don't need to worry about thread safety. So here you can just change this code to - 

```
metrics.computeIfAbsent(name, RuntimeMetric::new).addValue(value);
```

Reply:
Thanks for the suggestion. Done.